src/vllm_tgis_adapter/grpc/adapters.py

```diff
@@ -107,7 +107,10 @@ async def validate_adapters(
             )
 
             # Add to cache
-            adapter_store.adapters[adapter_id] = adapter_metadata
+            # We don't cache LoRA requests because the vllm api server
+            # maintains its own cache
+            if adapter_metadata.adapter_type != "LORA":
+                adapter_store.adapters[adapter_id] = adapter_metadata
```

Comment:
🤔 

In the case where we:
- Checked and saw that the REST server does not have this LoRA request in its cache, and
- Validated that this file path points to a lora adapter

We actually need to give this information to the REST server so that it can be responsible for creating and caching the LoRA request. If we don't, then we still end up in the situation where we've created a request for the adapter from two separate places, so we'll end up with two separate adapters loaded from the same data into the gpu cache in the engine. (And if we just drop caching altogether, we'll end up with N copies for N requests 😱 ) 

`vllm_server.state.openai_serving_models.load_lora_adapter` should do the caching on the REST side, I think

Reply:
Oh, I didn't realize we were loading it on the GPU as well, I assumed that request got sent over to vllm at some point 🤔